index.js

```diff
@@ -155,36 +162,45 @@ async function underPressure (fastify, opts) {
 
   function onRequest (req, reply, next) {
     if (checkMaxEventLoopDelay && eventLoopDelay > maxEventLoopDelay) {
-      sendError(reply, next)
+      handlePressure(req, reply, next, TYPE_EVENT_LOOP_DELAY, eventLoopDelay)
       return
     }
 
     if (checkMaxHeapUsedBytes && heapUsed > maxHeapUsedBytes) {
-      sendError(reply, next)
+      handlePressure(req, reply, next, TYPE_HEAP_USED_BYTES, heapUsed)
       return
     }
 
     if (checkMaxRssBytes && rssBytes > maxRssBytes) {
-      sendError(reply, next)
+      handlePressure(req, reply, next, TYPE_RSS_BYTES, rssBytes)
       return
     }
 
     if (!externalsHealthy) {
-      sendError(reply, next)
+      handlePressure(req, reply, next, TYPE_HEALTH_CHECK)
       return
     }
 
     if (checkMaxEventLoopUtilization && eventLoopUtilized > maxEventLoopUtilization) {
-      sendError(reply, next)
+      handlePressure(req, reply, next, TYPE_EVENT_LOOP_UTILIZATION, eventLoopUtilized)
       return
     }
 
     next()
   }
 
-  function sendError (reply, next) {
-    reply.status(SERVICE_UNAVAILABLE).header('Retry-After', retryAfter)
-    next(underPressureError)
+  function handlePressure (req, reply, next, type, value) {
+    if (typeof pressureHandler === 'function') {
+      const result = pressureHandler(req, reply, type, value)
+      if (result instanceof Promise) {
+        result.then(() => next()).catch(next)
+      } else {
```

Comment:
I think you should handle the case where result is something else of a promise.

Reply:
But how? What am I supposed to do with a non-Promise? Calling `reply.send()` with the object/string/etc?

Reply:
Either that or error the response

Reply:
done